### PR TITLE
Merging ambient occlusion into master

### PR DESCRIPTION
**Pull request by likangning93**
As noted [when we were finishing up the uniform grid](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/124#issuecomment-232474010), I took a look at the Node guide and added some documentation and cleanup.

`bakeAmbientOcclusion` should come after [`cache optimization`](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/80) in any pipeline configuration, so I didn't match it to the promise-based function [template](https://github.com/AnalyticalGraphicsInc/gltf-pipeline/issues/99) yet.

### CODE REVIEW

**Comment by pjcozzi** [comment]
**Exciting!**

@lilleyse please do the honors.

File: bin/gltf-pipeline.js [hunk 1]
```diff
@@ -55,6 +64,8 @@ var options = {
     quantize : quantize,
     encodeNormals : encodeNormals,
     compressTextureCoordinates : compressTextureCoordinates,
+    aoOptions : aoOptions,
+    imageProcess : defined(aoOptions) && aoOptions.toTexture,
```
Comment (lilleyse) [inline bin/gltf-pipeline.js]: I removed `imageProcess` in https://github.com/AnalyticalGraphicsInc/gltf-pipeline/pull/128, so you can delete this here.

File: lib/bakeAmbientOcclusion.js [hunk 2]
```diff
@@ -0,0 +1,1352 @@
...
+            value = 255.0 * (aoBuffer.samples[dataIdx] / sampleCount);
+            jimpAO.bitmap.data[dataIdx * 4 + 3] = value;
+        }
+    }
+    // Resize the data to match the goal resolution
+    jimpAO.resize(goalResolution, goalResolution, Jimp.RESIZE_BEZIER);
+}
+
+function cloneAndSetupMaterialTextureImage(options) {
+
```
Comment (lilleyse) [inline lib/bakeAmbientOcclusion.js]: Remove white space.

File: specs/lib/bakeAmbientOcclusionSpec.js [hunk 3]
```diff
@@ -0,0 +1,784 @@
...
+                    }
+                ]
+            }
+        }
+    };
+
+    beforeAll(function(done) {
+        var options = {
+            imageProcess: true
+        };
```
Comment (lilleyse) [inline specs/lib/bakeAmbientOcclusionSpec.js]: Also no need for `imageProcess` here.

**Comment by lilleyse** [comment]
After those comments this is ready to go.

**Comment by likangning93** [comment]
Updated!

**Comment by lilleyse** [comment]
test5

Looks great!

**Pull request merged**